### PR TITLE
Makes Plasma materials can reduce radiation strength from travelling through

### PR DESCRIPTION
**Pull request by warface1234455**
Reduces chernobyl station

# Document the changes in your pull request
Makes  plasma glass can **Reduce(Not completely remove)** radiation and rad particles from travelling through
<details><summary>radiation outside the plasma glass is reduced</summary>

image

</details>

# Wiki Documentation
Makes  plasma materials can reduce radiation and rad particles from travelling through

# Changelog

:cl:  warface1234455
rscadd: Makes plasma materials can reduce radiation and rad particles from travelling through
/:cl:

### CODE REVIEW

**Comment by CMOisLing** [comment]
Do rad balls still go through?

**Comment by VaelophisNyx** [comment]
it's a window so I assume so

**Comment by ShoziOffical** [comment]
nope

**Comment by ShoziOffical** [comment]
> Do rad balls still go through?

no

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
Should use rad_insulation component with some tweaks to ensure 100% blocking at amount = 0

**Comment by ShoziOffical** [comment]
> Should use rad_insulation component with some tweaks to ensure 100% blocking at amount = 0

it doesnt work lmaooo

**Comment by Bibby0110** [comment]
> > Should use rad_insulation component with some tweaks to ensure 100% blocking at amount = 0
> 
> it doesnt work lmaooo
 it does

**Comment by Bibby0110** [comment]
DNM until using component and reviewed for balance

**Comment by ShoziOffical** [comment]
> DNM until using component and reviewed for balance

I wouldn't make this pr just to block tritium rad or fusion chamber rad

**Comment by ShoziOffical** [comment]
smh bibby

**Comment by VaelophisNyx** [comment]
bibby moment

**Comment by VaelophisNyx** [comment]
> DNM until using component and reviewed for balance

Man you just love using that label huh

**Comment by Bibby0110** [comment]
> > DNM until using component and reviewed for balance
> 
> Man you just love using that label huh

sure do, else jamie inevitably merges it before it’s fixed

File: code/game/objects/structures/window.dm [hunk 1]
```diff
@@ -751,14 +756,20 @@
...
 	damage_deflection = 21 //The same as reinforced plasma windows.
 	level = 3
 	glass_type = /obj/item/stack/sheet/plastitaniumglass
 	glass_amount = 2
+	rad_insulation = RAD_FULL_INSULATION
+
+/obj/structure/window/plastitanium/CanAllowThrough(atom/movable/mover, turf/target)
+	. = ..()
+	if(istype(mover, /obj/item/projectile/energy/nuclear_particle))
+		return 0
```
Comment (Bibby0110) [inline code/game/objects/structures/window.dm]: TRUE/FALSE

Comment (ShoziOffical) [reply code/game/objects/structures/window.dm]: Isn't it the same

**Comment by Hopekz** [comment]
> > > DNM until using component and reviewed for balance
> > 
> > Man you just love using that label huh
> 
> sure do, else jamie inevitably merges it before it’s fixed

🍼👶

**Comment by Mqiib** [comment]
You realize the reason plasma glass had no rad insulation was because it specifically was for radioactive engines like the SM that needed radiation to reach the collectors while also having a pretty solid barrier to keep everything inside right?

**Comment by ShoziOffical** [comment]
> You realize the reason plasma glass had no rad insulation was because it specifically was for radioactive engines like the SM that needed radiation to reach the collectors while also having a pretty solid barrier to keep everything inside right?
 
If an object is on the plasma window tile it still can be contaminated and don't worry so much, i tested and engine still produce enough power for the station

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -567,10 +567,12 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			l.hallucination += power * config_hallucination_power * D
 			l.hallucination = clamp(l.hallucination, 0, 200)
 
-	for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
+	/*for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: ?

Comment (ShoziOffical) [reply code/modules/power/supermatter/supermatter.dm]: It literally bypasses the glass insulation thing, and the code isnt needed as there is radiation_pulse(max_rads) in supermatterm.dm that does the same thing

Comment (ShoziOffical) [reply code/modules/power/supermatter/supermatter.dm]: Any mob inside the range of 10 tiles from even behind 5 reinforced wall still can be fried to dead

**Comment by JamieD1** [comment]
Want @TheGamerdk To review

File: code/modules/power/supermatter/supermatter.dm (hunk 2 above)
Comment (Bibby0110) [reply code/modules/power/supermatter/supermatter.dm]: well then remove the code?

Comment (ShoziOffical) [reply code/modules/power/supermatter/supermatter.dm]: yeah

**Comment by ShoziOffical** [comment]
@TheGamerdk re-review

**Comment by cuackles** [comment]
why is /art required to review this ? i dont see diference in the iconbot or dmi files

**Comment by ShoziOffical** [comment]
> why is /art required to review this ? i dont see diference in the iconbot or dmi files

Github moment

**Review by Bibby0110 — APPROVED** [review]

**Pull request merged**